### PR TITLE
Weighted diagnostics

### PR DESCRIPTION
**Pull request by n-kall**
#### Summary

Adds weighted ESS, MCSE and Pareto diagnostics.

#### Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by paul-buerkner** [comment]
Thank you! @avehtari would you have time to review this PR?

**Comment by n-kall** [comment]
I should still add some documentation, but I think the core functionality is there (plus tests)

File: R/convergence.R [hunk 1]
```diff
@@ -484,28 +570,40 @@ mcse_sd <- function(x, ...) UseMethod("mcse_sd")
...
+    # The first order Taylor series approximation of variance of sd.
+    # Kenney and Keeping (1951, p. 141) write "...since fluctuations of
+    # any moment are of order N^{-1/2}, squares and higher powers of
+    # differentials of the moments can be neglected "
+    varsd <- varvar / Evar / 4
+    sqrt(varsd)
+
+  } else {
+
+    sims_c <- x - mean(x)
```
Comment (n-kall) [inline R/convergence.R:595]: This needs to be corrected to account for the equation for non-weighted mcse_sd above

**Comment by n-kall** [comment]
I think this can be merged into the weighted rvars branch, as @avehtari and I went through and reviewed this some time ago

**Comment by paul-buerkner** [comment]
Thanks. Merging now.

**Pull request merged**